pkg/monitortests/node/watchpods/compute_intervals.go

```diff
@@ -49,39 +51,55 @@ func intervalsFromEvents_PodChanges(events monitorapi.Intervals, beginning, end
+	// without having to parse legacy locator strings (which we're using in the map keys)
+	locatorKeyToLocator := map[string]monitorapi.Locator{}
 
 	for i := range input {
 		event := input[i]
-		pod := monitorapi.PodFrom(event.Locator)
-		if len(pod.Name) == 0 {
+		if _, ok := event.StructuredLocator.Keys[monitorapi.LocatorPodKey]; !ok {
 			continue
 		}
-		allPodTransitions[pod.ToLocator()] = append(allPodTransitions[pod.ToLocator()], event)
-		isRecognizedPodReason := monitorapi.PodLifecycleTransitionReasons.Has(monitorapi.ReasonFrom(event.Message))
 
-		container := monitorapi.ContainerFrom(event.Locator)
-		isContainer := len(container.ContainerName) > 0
-		isContainerLifecycleTransition := monitorapi.ContainerLifecycleTransitionReasons.Has(monitorapi.ReasonFrom(event.Message))
-		isContainerReadyTransition := monitorapi.ContainerReadinessTransitionReasons.Has(monitorapi.ReasonFrom(event.Message))
-		isKubeletReadinessCheck := monitorapi.KubeletReadinessCheckReasons.Has(monitorapi.ReasonFrom(event.Message))
+		// We have to strip out container, this needs to be just pod locator here
+		podLocator := monitorapi.PodFromLocator(event.StructuredLocator).ToLocator()
+		pls := podLocator.OldLocator()
+		locatorKeyToLocator[pls] = podLocator
+		reason := event.StructuredMessage.Reason
+		allPodTransitions[pls] = append(allPodTransitions[pls], event)
+		isRecognizedPodReason := monitorapi.PodLifecycleTransitionReasons.Has(reason)
+
+		var containerLocator monitorapi.Locator
+		var cls string
+		_, isContainer := event.StructuredLocator.Keys[monitorapi.LocatorContainerKey]
+		if isContainer {
+			containerLocator = monitorapi.ContainerFrom(event.StructuredLocator).ToLocator()
+			cls = containerLocator.OldLocator()
+			locatorKeyToLocator[cls] = containerLocator
+		}
+		isContainerLifecycleTransition := monitorapi.ContainerLifecycleTransitionReasons.Has(reason)
+		isContainerReadyTransition := monitorapi.ContainerReadinessTransitionReasons.Has(reason)
+		isKubeletReadinessCheck := monitorapi.KubeletReadinessCheckReasons.Has(reason)
 
 		switch {
 		case !isContainer && isRecognizedPodReason:
-			podToStateTransitions[pod.ToLocator()] = append(podToStateTransitions[pod.ToLocator()], event)
+			podToStateTransitions[pls] = append(podToStateTransitions[pls], event)
 
 		case isContainer && isContainerLifecycleTransition:
-			containerToLifecycleTransitions[container.ToLocator()] = append(containerToLifecycleTransitions[container.ToLocator()], event)
+			containerToLifecycleTransitions[cls] = append(containerToLifecycleTransitions[cls], event)
 
 		case isContainer && isContainerReadyTransition:
-			containerToReadinessTransitions[container.ToLocator()] = append(containerToReadinessTransitions[container.ToLocator()], event)
+			containerToReadinessTransitions[cls] = append(containerToReadinessTransitions[cls], event)
 
 		case isKubeletReadinessCheck:
```

Comment:
do we need `isContainer &&` here like the ones above that also use `cls`? or maybe `isKubeletReadinessCheck` imples that `isContainer == true`?

Reply:
Shouldn't no, this was it's original form, kubelet isn't run in a container (I believe it's an os level systemd service), and these intervals would not have a container key in their locator.